plugin-server/src/utils/db/db.ts

```diff
@@ -531,6 +549,57 @@ export class DB {
         return client ? kafkaMessages : updatedPerson
     }
 
+    public async updatePersonProperties(
+        teamId: number,
+        distinctId: string,
+        properties: Properties,
+        propertiesOnce: Properties,
+        timestamp: DateTime
+    ): Promise<void> {
+        if (properties.length == 0 && propertiesOnce.length == 0) {
+            return
+        }
+        let values: Array<any> = ['<id>', timestamp.toISO()]
+        const startIndex = values.length + 1
+        values = values.concat(getPersonPropertyUpdateValues(properties, PersonPropertyUpdateOperation.Set))
+        values = values.concat(getPersonPropertyUpdateValues(propertiesOnce, PersonPropertyUpdateOperation.SetOnce))
+        const propertyUpdateExpressions = getPersonPropertyUpdateExpressions(values, startIndex)
+        const query = `SELECT update_person_props($1, $2, ARRAY[ ${propertyUpdateExpressions} ] );`
+
+        let props: Properties = {}
+        let personFound: Person | undefined
+        await this.postgresTransaction(async (client) => {
+            personFound = await this.fetchPerson(teamId, distinctId)
+            if (!personFound) {
+                throw new Error(
+                    `Could not find person with distinct id "${distinctId}" in team "${teamId}" to update props`
+                )
+            }
+            values[0] = personFound.id
+            const updateResult = await client.query(query, values)
```

Comment:
How can I get the properties from here? 
They should be returned correctly based on tests in https://github.com/PostHog/posthog/pull/6649 which this PR depends on.

when I ran `tests/shared/db.test.ts -t "update non-existent single property"` I saw
```
PASS  tests/shared/db.test.ts (17.278 s)
  DB
    ○ skipped fetchAllActionsGroupedByTeam
    fetchGroupTypes() and insertGroupType()
      ○ skipped fetches group types that have been inserted
      ○ skipped handles conflicting by index when inserting and limits
      ○ skipped handles conflict by name when inserting
    updatePersonProperties
      ✓ update non-existent single property (839 ms)
      ○ skipped update non-existent property
      ○ skipped update with newer timestamp - rows [2-5]
      ○ skipped update with equal timestamp - rows [6-9]
      ○ skipped update with older timestamp - rows [10-13]

  console.log
    Result {
      command: 'SELECT',
      rowCount: 1,
      oid: null,
      rows: [ { update_person_props: '' } ],
      fields: [
        Field {
          name: 'update_person_props',
          tableID: 0,
          columnID: 0,
          dataTypeID: 2278,
          dataTypeSize: 4,
          dataTypeModifier: -1,
          format: 'text'
        }
      ],
      _parsers: [ [Function: noParse] ],
      _types: TypeOverrides {
        _types: {
          getTypeParser: [Function: getTypeParser],
          setTypeParser: [Function: setTypeParser],
          arrayParser: [Object],
          builtins: [Object]
        },
        text: {},
        binary: {}
      },
      RowCtor: null,
      rowAsArray: false
    }

      at src/utils/db/db.ts:580:21

  console.log
    [ { update_person_props: '' } ]

      at src/utils/db/db.ts:581:21

  console.log
    { update_person_props: '' }

      at src/utils/db/db.ts:585:21

  console.log
    {}

      at DB.updatePersonProperties (src/utils/db/db.ts:588:17)
```

Reply:
need to have the right branch as base

Reply:
ah I see now you added the migration here (great call btw)